pkg/engine/virtualmachinescalesets.go

```diff
@@ -721,6 +722,7 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 			registry = `mcr.microsoft.com 443`
 		}
 		outBoundCmd = `retrycmd_if_failure() { r=$1; w=$2; t=$3; shift && shift && shift; for i in $(seq 1 $r); do timeout $t ${@}; [ $? -eq 0  ] && break || if [ $i -eq $r ]; then return 1; else sleep $w; fi; done }; ERR_OUTBOUND_CONN_FAIL=50; retrycmd_if_failure 50 1 3 ` + ncBinary + ` -vz ` + registry + ` || exit $ERR_OUTBOUND_CONN_FAIL;`
+		outBoundK8sCmd = `;ERR_K8S_API_SERVER_CONN_FAIL=51; retrycmd_if_failure 50 1 3 ` + ncBinary + ` -vz variables('kubernetesAPIServerIP') 443 || exit $ERR_K8S_API_SERVER_CONN_FAIL;`
```

Comment:
I think this might be easier to maintain in the CSE script itself. Like, at the bottom of `cse_main.sh` you could add something like:

```
{{IsHostedMaster}}
retrycmd_if_failure 50 1 3 nc  -vz $APISERVER_IP 443 || exit $ERR_K8S_API_SERVER_CONN_FAIL
{{end}}
```

Using the above you'd need to add a `APISERVER_IP` env variable (see under `pkg/engine/armvariables.go`).

Do you have any thoughts @CecileRobertMichon ?

Reply:
Does that mean we'll be also doing this for master?

Reply:
No. `IsHostedMaster` will return false for *all* CSE contexts where there is a master VM (e.g., an AKS Engine scenario)

Reply:
+1 on Jack's comment

Reply:
ok I'm moving it to cse